extension/lib/shopgate/CartExtensionPipeline.js

```diff
@@ -120,11 +120,19 @@ class ShopgateCartExtensionPipeline {
     await this._bigCommerceCartRepository.deleteProductFromCart(cartItemIds)
   }
 
+  /**
+   * @returns {Promise<void>}
+   */
+  destroyCart () {
+    return this._bigCommerceCartRepository.destroy()
+  }
+
   /**
    * @param {PipelineContext} context
+   * @param {boolean} [forceDeviceStorage = false] force using device storage, even if user is logged in
    * @returns {ShopgateCartExtensionPipeline}
    */
-  static create (context) {
+  static create (context, forceDeviceStorage = false) {
```

Comment:
What do you think if we'd, instead of introducing new parameter here, introduce new factory method, say createUser, we'd avoid adding complexity to existing create method and add semantical context around it.

Reply:
I think the added complexity is very minimal and can be neglected in this case, especially when compared to the redundant code, we'd otherwise have.

Reply:
Redundancy is sometimes fine to have, and for factories and factory methods I'd say it is usual. What matters more are dependencies. Adding a parameter to the create method will make all consumers dependant on the new parameter with an implicit value of false. It might be easier to talk about this.

Reply:
done